array-destructuring/exercise-3/exercise.js

```diff
@@ -6,3 +6,14 @@ let order = [
   { itemName: "Hot Coffee", quantity: 2, unitPrice: 1.0 },
   { itemName: "Hash Brown", quantity: 4, unitPrice: 0.4 },
 ];
+let grandTotal = 0;
+
+console.log("QTY\tITEM\t\t\t\tTOTAL");
+
+order.forEach(({ quantity, itemName, unitPrice }) => {
+  const total = unitPrice * quantity;
+  grandTotal += total;
+  console.log(`${quantity}\t${itemName}\t\t${total.toFixed(2)}`);
+});
+
+console.log("\nTotal: ", grandTotal.toFixed(2));
```

Comment:
This is a reasonable approach!

An alternative might be to separate out the receipt lines from the total calculation using a reduce function to calculate the total:

```js
console.log("QTY\tITEM\t\t\t\tTOTAL");

order.forEach(({ quantity, itemName, unitPrice }) => {
  const total = unitPrice * quantity;
  console.log(`${quantity}\t${itemName}\t\t${total.toFixed(2)}`);
});

const grandTotal = order.reduce(
  (sum, { quantity, unitPrice }) => sum + quantity * unitPrice,
  0
);

console.log("\nTotal: ", grandTotal.toFixed(2));
```

The reduce function is a little annoying to write but object destructuring does make it a little easier, and it means that we're keeping our logic for the item lines and the total separated.